assets/src/components/parts/janus-capi-iframe/ExternalActivity.tsx

```diff
@@ -318,6 +318,56 @@ const ExternalActivity: React.FC<any> = (props) => {
     }
   };
 
+  const getInterestedVars = (vars: any) => {
+    const interested = Object.keys(vars).filter((ms: any) => {
+      const isMine = ms.indexOf(`stage.${id}.`) === 0;
+      if (!isMine) {
+        return false;
```

Comment:
this is a fairly big method with multiple return types (boolean here, object elsewhere) that makes it difficult to reason about. perhaps some of it can be factored into smaller functions. also if there can be one return type that would help

Reply:
nevermind the multiple return types, I see that is a filter, maybe that shows however that its big and easy to get lost :D